src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/WebClientConfiguration.kt

```diff
@@ -3,16 +3,50 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.config
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.security.oauth2.client.AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.InMemoryReactiveOAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ClientRegistration
+import org.springframework.security.oauth2.client.registration.InMemoryReactiveClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServerOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.security.oauth2.core.AuthorizationGrantType
 import org.springframework.web.reactive.function.client.WebClient
 
 @Configuration
 class WebClientConfiguration(
   @Value("\${hmpps.auth.url}") private val oauthApiUrl: String,
-  private val webClientBuilder: WebClient.Builder
+  @Value("\${hmpps.prison.api.url}") private val prisonApiUrl: String,
+  @Value("\${oauth.client.token-url}") private val oauthTokenUrl: String,
+  @Value("\${oauth.client.id}") private val oauthClientId: String,
+  @Value("\${oauth.client.secret}") private val oauthSecret: String,
 ) {
 
   @Bean
   fun oauthApiHealthWebClient(): WebClient {
-    return webClientBuilder.baseUrl(oauthApiUrl).build()
+    return WebClient.builder().baseUrl(oauthApiUrl).build()
+  }
+
+  @Bean
```

Comment:
There are several ways to configure the oauth configuration. This was the only route I had success with, but I'm open to other methods if deeded preferable.

Reply:
we have a slightly different way in WPIP and connecting to basm - we're not using reactive clients though. 
```kt
  @Bean
  fun authorizedClientManager(
    clientRegistrationRepository: ClientRegistrationRepository?,
    oAuth2AuthorizedClientService: OAuth2AuthorizedClientService?
  ): OAuth2AuthorizedClientManager? {
    val authorizedClientProvider = OAuth2AuthorizedClientProviderBuilder.builder().clientCredentials().build()
    val authorizedClientManager =
      AuthorizedClientServiceOAuth2AuthorizedClientManager(clientRegistrationRepository, oAuth2AuthorizedClientService)
    authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider)
    return authorizedClientManager
  }

  @Bean
  fun basmApiWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
    oauth2Client.setDefaultClientRegistrationId("basm-api")

    return WebClient.builder()
      .baseUrl(basmRootUri)
      .apply(oauth2Client.oauth2Configuration())
      .exchangeStrategies(
        ExchangeStrategies.builder()
          .codecs { configurer ->
            configurer.defaultCodecs()
              .maxInMemorySize(-1)
          }
          .build()
      )
      .build()
  }
```

So we don't have to manually create the client registration - that's just picked up from config automagically.

Reply:
https://github.com/ministryofjustice/hmpps-welcome-people-into-prison-api/blob/main/src/main/resources/application.yml#L56-L62

Reply:
I was able to refactor the code to reflect the referenced code. I think my initial issues were around the yaml config being slightly out of whack. It works now. I'll push the changes.